vision/cloud-client test failures (Flakey tests?) upgrade to libraries-BOM 9.1.0

Found in #3555 

```
------------------------------------------------------------
- testing vision/cloud-client
------------------------------------------------------------
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.368 s <<< FAILURE! - in com.example.vision.DetectLandmarksGcsTest
[ERROR] com.example.vision.DetectLandmarksGcsTest.testLandmarksUrl  Time elapsed: 2.696 s  <<< FAILURE!
expected to contain:
    palace of fine arts
but was:
    error: we can not access the url currently. please download the content and pass it in.

	at com.example.vision.DetectLandmarksGcsTest.testLandmarksUrl(DetectLandmarksGcsTest.java:61)

[ERROR] Failures:
[ERROR]   DetectLandmarksGcsTest.testLandmarksUrl:61 expected to contain:
    palace of fine arts
but was:
    error: we can not access the url currently. please download the content and pass it in.

[ERROR] Tests run: 26, Failures: 1, Errors: 0, Skipped: 0
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M4:test (default-test) on project vision-google-cloud-samples: There are test failures.
```

--- a/vision/cloud-client/src/main/java/com/example/vision/snippets/DetectLandmarksUrl.java
+++ b/vision/cloud-client/src/main/java/com/example/vision/snippets/DetectLandmarksUrl.java
@@ -40,7 +40,7 @@ public class DetectLandmarksUrl {
   }
 
   // Detects landmarks in the specified URI.
-  public static void detectLandmarksUrl(String uri) throws IOException {
+  public static void detectLandmarksUrl(String uri) throws IOException, StatusRuntimeException {
     List<AnnotateImageRequest> requests = new ArrayList<>();
 
     ImageSource imgSource = ImageSource.newBuilder().setImageUri(uri).build();

--- a/vision/cloud-client/src/main/java/com/example/vision/snippets/DetectLandmarksUrl.java
+++ b/vision/cloud-client/src/main/java/com/example/vision/snippets/DetectLandmarksUrl.java
@@ -32,7 +32,6 @@ import java.util.ArrayList;
 import java.util.List;
 
 public class DetectLandmarksUrl {
-  private static final String ASSET_BUCKET = "cloud-samples-data";
 
   public static void detectLandmarksUrl() throws IOException {
     // TODO(developer): Replace these variables before running the sample.

--- a/vision/cloud-client/src/main/java/com/example/vision/snippets/DetectLandmarksUrl.java
+++ b/vision/cloud-client/src/main/java/com/example/vision/snippets/DetectLandmarksUrl.java
@@ -25,11 +25,14 @@ import com.google.cloud.vision.v1.Image;
 import com.google.cloud.vision.v1.ImageAnnotatorClient;
 import com.google.cloud.vision.v1.ImageSource;
 import com.google.cloud.vision.v1.LocationInfo;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 
 public class DetectLandmarksUrl {
+  private static final String ASSET_BUCKET = "cloud-samples-data";
 
   public static void detectLandmarksUrl() throws IOException {
     // TODO(developer): Replace these variables before running the sample.
@@ -58,7 +61,7 @@ public class DetectLandmarksUrl {
       for (AnnotateImageResponse res : responses) {
         if (res.hasError()) {
           System.out.format("Error: %s%n", res.getError().getMessage());
-          return;
+          throw new StatusRuntimeException(Status.UNAVAILABLE);
         }
 
         // For full list of available annotations, see http://g.co/cloud/vision/docs

--- a/vision/cloud-client/src/test/java/com/example/vision/DetectLandmarksGcsTest.java
+++ b/vision/cloud-client/src/test/java/com/example/vision/DetectLandmarksGcsTest.java
@@ -20,6 +20,7 @@ import static com.google.common.truth.Truth.assertThat;
 
 import com.example.vision.snippets.DetectLandmarksGcs;
 import com.example.vision.snippets.DetectLandmarksUrl;
+import io.grpc.StatusRuntimeException;
 import java.io.ByteArrayOutputStream;
 import java.io.PrintStream;
 import org.junit.After;
@@ -63,11 +64,21 @@ public class DetectLandmarksGcsTest {
 
   @Test
   public void testLandmarksGcs() throws Exception {
-    // Act
-    DetectLandmarksGcs.detectLandmarksGcs("gs://" + ASSET_BUCKET + "/vision/landmark/pofa.jpg");
+    int tryCount = 0;
+    int maxTries = 3;
 
-    // Assert
-    String got = bout.toString().toLowerCase();
-    assertThat(got).contains("palace of fine arts");
+    while (tryCount < maxTries) {
+      try {
+        // Act
+        DetectLandmarksGcs.detectLandmarksGcs("gs://" + ASSET_BUCKET + "/vision/landmark/pofa.jpg");
+        // Assert
+        String got = bout.toString().toLowerCase();
+        assertThat(got).contains("palace of fine arts");
+        break;
+      } catch (StatusRuntimeException ex) {
+        System.out.println("Retrying...");
+        tryCount++;
+      }
+    }
   }
 }

